idea/src/org/jetbrains/kotlin/idea/inspections/ImplicitThisInspection.kt

```diff
@@ -53,7 +54,10 @@ class ImplicitThisInspection : AbstractKotlinInspection() {
             val scope = reference.getResolutionScope(context) ?: return
 
             val descriptor = context[BindingContext.REFERENCE_TARGET, reference] as? CallableDescriptor ?: return
-            val receiverDescriptor = descriptor.extensionReceiverParameter ?: descriptor.dispatchReceiverParameter ?: return
+            val receiverDescriptor = descriptor.extensionReceiverParameter
+                    ?: descriptor.dispatchReceiverParameter
+                    ?: expression.getCallableDescriptor()?.extensionReceiverParameter
```

Comment:
Should be Ok. I'll check whether `descriptor` can be just changed with `getCallableDescriptor()` here.